stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -812,15 +812,10 @@ function eigen(D::Diagonal; permute::Bool=true, scale::Bool=true, sortby::Union{
     end
     Td = Base.promote_op(/, eltype(D), eltype(D))
     λ = eigvals(D)
+    evecs = Diagonal{eltype(λ)}(ones(length(λ)))
```

Comment:
I suspect this needs to replace the old line 823, because when you want to sort by eigenvalues, then this will fail with a Diagonal matrix.